### PR TITLE
core: update .net7.0

### PR DESCRIPTION
**Pull request by ngosang**
TODO:
* Remove or replace AutoMapper => They are not compatible with .Net/Mono anymore =>  #13879
* Profile the memory usage. it has increased a lot vs .net6.0

This error when I search something
```
01-07 22:01:04 Error System.TypeInitializationException: The type initializer for 'Jackett.Common.Utils.MapperUtil' threw an exception. System.TypeInitializationException: The type initializer for 'Jackett.Common.Utils.MapperUtil' threw an exception.
 ---> System.ArgumentException: GenericArguments[0], 'System.Int32', on 'T MaxFloat[T](System.Collections.Generic.IEnumerable`1[T])' violates the constraint of type 'T'.
 ---> System.Security.VerificationException: Method System.Linq.Enumerable.MaxFloat: type argument 'System.Int32' violates the constraint of type parameter 'T'.
   at System.RuntimeMethodHandle.GetStubIfNeeded(RuntimeMethodHandleInternal method, RuntimeType declaringType, RuntimeType[] methodInstantiation)
   at System.Reflection.RuntimeMethodInfo.MakeGenericMethod(Type[] methodInstantiation)
   --- End of inner exception stack trace ---
   at System.RuntimeType.ValidateGenericArguments(MemberInfo definition, RuntimeType[] genericArguments, Exception e)
   at System.Reflection.RuntimeMethodInfo.MakeGenericMethod(Type[] methodInstantiation)
   at AutoMapper.TypeDetails.<>c__DisplayClass28_1.<BuildPublicNoArgExtensionMethods>b__10(MethodInfo extensionMethod)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Enumerable.ConcatIterator`1.MoveNext()
   at System.Linq.Enumerable.SelectManyIterator[TSource,TCollection,TResult](IEnumerable`1 source, Func`2 collectionSelector, Func`3 resultSelector)+MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Linq.Enumerable.UnionIterator`1.FillSet()
   at System.Linq.Enumerable.UnionIterator`1.ToArray()
   at AutoMapper.TypeDetails.BuildPublicNoArgExtensionMethods(IEnumerable`1 sourceExtensionMethodSearch)
   at AutoMapper.TypeDetails..ctor(Type type, ProfileMap config)
   at AutoMapper.ProfileMap.TypeDetailsFactory(Type type)
   at AutoMapper.Internal.LockingConcurrentDictionary`2.<>c__DisplayClass2_1.<.ctor>b__1()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at AutoMapper.Internal.LockingConcurrentDictionary`2.GetOrAdd(TKey key)

```

### CODE REVIEW

**Comment by ilike2burnthing** [comment]
Resolved a conflict caused by https://github.com/Jackett/Jackett/commit/d0ebdea1643f330e0519ba0d06a7a84ba12d9173

**Comment by Uruknara** [comment]
when merging all of the pr ?

**Comment by ilike2burnthing** [comment]
When @ngosang is free. There's no rush.

**Comment by Uruknara** [comment]
> 

yes your right im just hopping search are gonna be such faster with net.7 and all of the other pr

**Comment by Qstick** [comment]
> yes your right im just hopping search are gonna be such faster with net.7 and all of the other pr

Probably not the case that anything will be MUCH faster with net7. 

Net7 also is a bit of a monster with resource usage compared to net6 in our testing (ngosang also mentioned similar above).

**Comment by goremykin** [comment]
> > yes your right im just hopping search are gonna be such faster with net.7 and all of the other pr
> 
> Probably not the case that anything will be MUCH faster with net7.
> 
> Net7 also is a bit of a monster with resource usage compared to net6 in our testing (ngosang also mentioned similar above).

And that surprised me a lot, because .net team made a lot of [cool optimizations](https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7/) in .net 7. So I expected at least the same memory usage or less...
They released some minor bugfix releases, I will profile with the latest one.

**Comment by goremykin** [comment]
I didn't profile much, but in my case with .net 7 it consumes a bit more memory (5-8% in my scenario) than with .net 6.
@Qstick, could you provide a bit more details how to profile to get the same huge usage?

Btw, Rider showed a warning about vulnerable dependencies:
https://devhub.checkmarx.com/cve-details/CVE-2019-0548/
https://devhub.checkmarx.com/cve-details/CVE-2019-0815/

**Comment by Qstick** [comment]
@goremykin 

The comment for Jackett was in this original post here

```Profile the memory usage. it has increased a lot vs .net6.0```

My comment was around the core arr programs and services, we've decided against moving to net7 and will move to net8 from net6 as we discussed in the Sonarr thread.

**Comment by txtsd** [comment]
> My comment was around the core arr programs and services, we've decided against moving to net7 and will move to net8 from net6 as we discussed in the Sonarr thread.

Is this still the case, or are we sticking with .NET 6.x for the foreseeable future?

**Comment by ilike2burnthing** [comment]
dotNET 8 is out now, so if there's to be any change I'd assume it would be to that.

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ A third-party Golang SDK for Jackett is available from [webtor-io/go-jackett](ht
 
 #### Supported Systems
 * Windows 7 SP1 or greater
-* Linux [supported operating systems here](https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md#linux)
+* Linux [supported operating systems here](https://github.com/dotnet/core/blob/main/release-notes/7.0/supported-os.md#linux)
```
Comment (PublisherApplications) [inline README.md:23]: Thank you for fixing the outdated content

File: README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ A third-party Golang SDK for Jackett is available from [webtor-io/go-jackett](ht
 
 #### Supported Systems
 * Windows 7 SP1 or greater
-* Linux [supported operating systems here](https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md#linux)
```
Comment (PublisherApplications) [inline README.md:23]: Thank you for fixing outdated content

File: README.md [hunk 3]
```diff
@@ -859,21 +859,21 @@ git clone https://github.com/Jackett/Jackett.git
 cd Jackett/src
 
 # dotnet core version
-dotnet publish Jackett.Server -f net6.0 --self-contained -r osx-x64 -c Debug # takes care of everything
-./Jackett.Server/bin/Debug/net6.0/osx-x64/jackett # run jackett
+dotnet publish Jackett.Server -f net7.0 --self-contained -r osx-x64 -c Debug # takes care of everything
```
Comment (PublisherApplications) [inline README.md:862]: Thank you for fixing the outdated content

File: README.md [hunk 4]
```diff
@@ -859,21 +859,21 @@ git clone https://github.com/Jackett/Jackett.git
...
+./Jackett.Server/bin/Debug/net7.0/osx-x64/jackett # run jackett
```
Comment (PublisherApplications) [inline README.md:863]: Thank you for fixing the outdated content.

File: README.md [hunk 5]
```diff
@@ -859,21 +859,21 @@ git clone https://github.com/Jackett/Jackett.git
...
 ```
 
 ### Linux
 
 
 ```bash
-sudo apt install nuget msbuild dotnet-sdk-6.0 # install build tools (Debian/ubuntu)
+sudo apt install nuget msbuild dotnet-sdk-7.0 # install build tools (Debian/ubuntu)
```
Comment (PublisherApplications) [inline README.md:870]: Thank you for fixing the outdated content.

File: README.md [hunk 6]
```diff
@@ -859,21 +859,21 @@ git clone https://github.com/Jackett/Jackett.git
...
 git clone https://github.com/Jackett/Jackett.git
 cd Jackett/src
 
 # dotnet core version
-dotnet publish Jackett.Server -f net6.0 --self-contained -r linux-x64 -c Debug # takes care of everything
-./Jackett.Server/bin/Debug/net6.0/linux-x64/jackett # run jackett
+dotnet publish Jackett.Server -f net7.0 --self-contained -r linux-x64 -c Debug # takes care of everything
```
Comment (PublisherApplications) [inline README.md:875]: Thank you for fixing the outdated content...

File: README.md [hunk 7]
```diff
@@ -859,21 +859,21 @@ git clone https://github.com/Jackett/Jackett.git
...
+./Jackett.Server/bin/Debug/net7.0/linux-x64/jackett # run jackett
```
Comment (PublisherApplications) [inline README.md:876]: Thank you for fixing the outdated content..

File: azure-pipelines.yml [hunk 8]
```diff
@@ -2,12 +2,12 @@
 name: $(majorVersion).$(minorVersion).$(patchVersion)
 variables:
   majorVersion: 0
-  minorVersion: 20
+  minorVersion: 21
```
Comment (PublisherApplications) [inline azure-pipelines.yml:5]: Thank you for fixing the outdated content

File: azure-pipelines.yml [hunk 9]
```diff
@@ -2,12 +2,12 @@
...
   patchVersion: $[counter(variables['minorVersion'], 1)]  # this will reset when we bump minor
   jackettVersion: $(majorVersion).$(minorVersion).$(patchVersion)
   buildConfiguration: Release
-  netCoreFramework: net6.0
-  netCoreSdkVersion: 6.0.x
+  netCoreFramework: net7.0
```
Comment (PublisherApplications) [inline azure-pipelines.yml:9]: Thank you for fixing the outdated content.

File: azure-pipelines.yml [hunk 10]
```diff
@@ -2,12 +2,12 @@
...
+  netCoreSdkVersion: 7.0.x
```
Comment (PublisherApplications) [inline azure-pipelines.yml:10]: Thank you for fixing the outdated content.

File: src/Jackett.Common/Jackett.Common.csproj [hunk 11]
```diff
@@ -27,7 +27,7 @@
     <PackageReference Include="polly" Version="7.2.3" />
     <PackageReference Include="SharpZipLib" Version="1.4.1" />
     <PackageReference Include="System.IO.FileSystem.AccessControl" Version="5.0.0" />
-    <PackageReference Include="System.ServiceProcess.ServiceController" Version="6.0.0" />
+    <PackageReference Include="System.ServiceProcess.ServiceController" Version="7.0.0" />
```
Comment (PublisherApplications) [inline src/Jackett.Common/Jackett.Common.csproj:30]: Thank you for fixing the outdated content.

File: src/Jackett.IntegrationTests/Jackett.IntegrationTests.csproj [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net462</TargetFrameworks>
+    <TargetFrameworks>net7.0;net462</TargetFrameworks>
```
Comment (PublisherApplications) [inline src/Jackett.IntegrationTests/Jackett.IntegrationTests.csproj:4]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 13]
```diff
@@ -1,20 +1,20 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net462</TargetFrameworks>
+    <TargetFrameworks>net7.0;net462</TargetFrameworks>
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:4]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 14]
```diff
@@ -1,20 +1,20 @@
...
     <ApplicationIcon>jackett.ico</ApplicationIcon>
     <OutputType>Exe</OutputType>
     <NoWarn></NoWarn>
     <ServerGarbageCollection>false</ServerGarbageCollection>
     <DefineConstants Condition="'$(RuntimeIdentifier)' == 'linux-musl-x64' or '$(RuntimeIdentifier)' == 'linux-musl-arm' or '$(RuntimeIdentifier)' == 'linux-musl-arm64'">ISLINUXMUSL</DefineConstants>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net6.0|AnyCPU'">
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0|AnyCPU'">
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:12]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 15]
```diff
@@ -1,20 +1,20 @@
...
     <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
     <WarningsAsErrors />
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net6.0|AnyCPU'">
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0|AnyCPU'">
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:17]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 16]
```diff
@@ -32,11 +32,11 @@
     </Otherwise>
   </Choose>
 
-  <!-- Conditionally obtain references for the .NET 6.0 target -->
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.12" />
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
-    <PackageReference Include="System.Security.Cryptography.ProtectedData" Version="6.0.0" />
+  <!-- Conditionally obtain references for the .NET 7.0 target -->
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net7.0' ">
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:36]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 17]
```diff
@@ -32,11 +32,11 @@
     </Otherwise>
   </Choose>
 
-  <!-- Conditionally obtain references for the .NET 6.0 target -->
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.12" />
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
-    <PackageReference Include="System.Security.Cryptography.ProtectedData" Version="6.0.0" />
+  <!-- Conditionally obtain references for the .NET 7.0 target -->
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:35]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 18]
```diff
@@ -32,11 +32,11 @@
...
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net7.0' ">
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="7.0.1" />
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:37]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 19]
```diff
@@ -32,11 +32,11 @@
...
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="7.0.0" />
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:38]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 20]
```diff
@@ -32,11 +32,11 @@
...
+    <PackageReference Include="System.Security.Cryptography.ProtectedData" Version="7.0.0" />
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:39]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 21]
```diff
@@ -56,11 +56,11 @@
     <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="8.0.0" />
     <PackageReference Include="AutoMapper" Version="10.1.1" />
     <PackageReference Include="CommandLineParser" Version="2.9.1" />
-	<PackageReference Include="Mono.Posix" Version="7.1.0-final.1.21458.1" />
+	  <PackageReference Include="Mono.Posix" Version="7.1.0-final.1.21458.1" />
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:59]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 22]
```diff
@@ -56,11 +56,11 @@
...
     <PackageReference Include="NLog" Version="5.1.1" />
     <PackageReference Include="NLog.Web.AspNetCore" Version="5.2.1" />
-    <PackageReference Include="System.ServiceProcess.ServiceController" Version="6.0.0" />
-    <PackageReference Include="System.Text.Encoding.CodePages" Version="6.0.0" />
+    <PackageReference Include="System.ServiceProcess.ServiceController" Version="7.0.0" />
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:62]: Thank you for fixing the outdated content.

File: src/Jackett.Server/Jackett.Server.csproj [hunk 23]
```diff
@@ -56,11 +56,11 @@
...
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="7.0.0" />
```
Comment (PublisherApplications) [inline src/Jackett.Server/Jackett.Server.csproj:63]: Thank you for fixing the outdated content.

File: src/Jackett.Service/Jackett.Service.csproj [hunk 24]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0-windows</TargetFramework>
+    <TargetFramework>net7.0-windows</TargetFramework>
```
Comment (PublisherApplications) [inline src/Jackett.Service/Jackett.Service.csproj:4]: Thank you for fixing the outdated content.

File: src/Jackett.Test/Jackett.Test.csproj [hunk 25]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net462</TargetFrameworks>
+    <TargetFrameworks>net7.0;net462</TargetFrameworks>
```
Comment (PublisherApplications) [inline src/Jackett.Test/Jackett.Test.csproj:4]: Thank you for fixing the outdated content.

File: src/Jackett.Test/Jackett.Test.csproj [hunk 26]
```diff
@@ -25,14 +25,14 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="FluentAssertions" Version="6.8.0" />
-    <PackageReference Include="Microsoft.AspNetCore.DataProtection" Version="6.0.12" />
+    <PackageReference Include="Microsoft.AspNetCore.DataProtection" Version="7.0.1" />
```
Comment (PublisherApplications) [inline src/Jackett.Test/Jackett.Test.csproj:28]: Thank you for fixing the outdated content....

File: src/Jackett.Test/Jackett.Test.csproj [hunk 27]
```diff
@@ -25,14 +25,14 @@
...
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
     <PackageReference Include="MSTest.TestAdapter" Version="3.0.2" />
     <PackageReference Include="MSTest.TestFramework" Version="3.0.2" />
     <PackageReference Include="NUnit" Version="3.13.3" />
     <PackageReference Include="NUnit.ConsoleRunner" Version="3.16.1" />
     <PackageReference Include="NUnit3TestAdapter" Version="4.3.1" />
-    <PackageReference Include="System.Text.Encoding.CodePages" Version="6.0.0" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="7.0.0" />
```
Comment (PublisherApplications) [inline src/Jackett.Test/Jackett.Test.csproj:35]: Thank you for fixing the outdated content.,.

File: src/Jackett.Tray/Jackett.Tray.csproj [hunk 28]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0-windows</TargetFramework>
+    <TargetFramework>net7.0-windows</TargetFramework>
```
Comment (PublisherApplications) [inline src/Jackett.Tray/Jackett.Tray.csproj:4]: Thank you for fixing the outdated content.

File: src/Jackett.Updater/Jackett.Updater.csproj [hunk 29]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net462</TargetFrameworks>
+    <TargetFrameworks>net7.0;net462</TargetFrameworks>
```
Comment (PublisherApplications) [inline src/Jackett.Updater/Jackett.Updater.csproj:4]: Thank you for fixing the outdated content.

File: src/Jackett.Updater/Jackett.Updater.csproj [hunk 30]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net462</TargetFrameworks>
```
Comment (PublisherApplications) [inline src/Jackett.Updater/Jackett.Updater.csproj:4]: These little changes are perfect to make sure you are a great collaborator

**Review by PublisherApplications — APPROVED** [review]
### Why did my changes get approved?
After reviewing your changes everything was just changes for the version And this is approved in my eyes, There may be second opinions but, You did pretty good with these changes.

**Comment by ilike2burnthing** [comment]
Not sure if this is a bot, but if you're a human then please never leave a line by line copy and paste 'review', especially on a year old PR. This was a waste of time, mine and yours.

**Comment by txtsd** [comment]
> dotNET 8 is out now, so if there's to be any change I'd assume it would be to that. No plans here though.

Yes that's what I meant. Brain wasn't braining so my sentence was ambiguous.

If no one else is working on it yet, I'll try and see if I can do the port to dotNET 8.

**Comment by ilike2burnthing** [comment]
More high memory usage issues?

**Pull request closed without merging**